### DISCUSSION TITLE
Error: U2 designator don't exist in the BOM file.

### DISCUSSION THREAD

**Discussion by pfundstein** [Dakkaron/Fairberry#15] [Q&A] [answered]
Hello, 

first of all, this is an awesome project! 

I am trying to get along with the PCB page as written in the documentation (https://github.com/Dakkaron/Fairberry/blob/main/Documentation/Hardware_Fairberry_Mainboard.md). I am uploading the gerber-zip file, activate "PCB assembly" and upload the two other files...  

Now the webpage gives me an error: 
**The below parts won't be assembled due to data missing.
U2 designator don't exist in the BOM file.**

Honestly, this is not really where I have experience, but I want to try my best to create the "device". Sadly I don't know what to do about this error. What has changed? Have I done something wrong?

Maybe someone can help me on that.

All the best,
X

**Comment by pfundstein** [comment]
Or is this an issue at all?

**Reply by MantleFounder** [reply]
I got the same error as well.

**Reply by pfundstein** [reply]
Were you able to build the keyboard case successfully?

**Reply by Dakkaron** [reply, marked as answer]
Sorry for the late answer.

If you are building a v0.1.1 or v.0.2.0 PCB, just remove the line with the U2 designator from the CPL file.

JLCPCB used to be fine with CPL lines that didn't have anything in the BOM, but apparently they aren't any more.

Alternatively, you can go for a v0.3.0 PCB. I just published a pre-release for v0.3.0. It's pre-release since my order is still shipping, so I couldn't test it so far.

If you want to risk it, you can order that, or wait until I get my order and confirm it's working.

**Reply by pfundstein** [reply]
Alright. It seems you also can ignore the warning at JLCPCB but they are out of stock with some items. Or at least last week they were, I have not checked today.

[Version 0.3.0](https://github.com/Dakkaron/Fairberry/pull/16) sounds sweet. I just hate the virtual keyboards and I'm still crying for my old BlackBerry KeyOne - the new "Clicks" keyboard for iPhone is cool but I don't want to buy an iPhone. I have a Motorola ThinkPhone right now. I honestly don't have any experience with electronics like this but I seriously want to try to build this keyboard case - gaining the experience while doing it. Basically. I mean I did some really easy soldering but that's it. 

Off-topic:

I have some ideas in mind on this project. 

Do you think there is enough space to use something like a [USB-C splitter](https://www.ebay.de/itm/235373599458?itmmeta=01HPFX0T75BJ73B9VSFR7DNHXZ&hash=item36cd5c1ae2:g:0GQAAOSw-ElllkxH&itmvar=enc%3AAQAIAAAA4GHfruRMBJTPOnFlJlssji1TFVcmKtTHQfmVzcLfr29xyYMWWsdjWYd8j8TWGhmrSliv09J5QNk%2FGOKde3%2BMQBGPosye4WQH8x8S1Kon3Ndv94j2%2Bguj5RCGUuGJa2XVPCd3nndS9dXEem0KaDmdvL3dzdB2mWf3MI54S9UA9i9DHOIOKY2y3K9Ap1DSJI3okXLmrMNRT4%2FSFUV3kzz8yvkg%2BrRRty56x0nHpaQPXUDH2scfXd7p08ej46wKopTS5%2F0dGcc0Yzjv%2FPe8oRIxDiSwriB6qhUVMu%2BcUkwdS%2Fhi%7Ctkp%3ABk9SR_Kjg_2zYw) so we can have the keyboard on the device and still have a USB-C port for charging? That would be awesome. 

By the way, when you have the keyboard on the device, how is the sound quality on a phone call? Because the mic sits usually at the bottom of the device and we basically cover it with the keyboard case.

**Reply by Dakkaron** [reply]
I am still waiting for them to source the keyboard connectors I ordered there for v0.3.0. I'll probably just order a set without and hand-solder them to get it quicker.

A small USB-C splitter could work, but you'd need to check that it's in fact capable of outputting USB 2.0. Many of these only work for USB-C analog audio passthrough.

I found one device that splits out an USB 2.0 OTG port, a pass-through charging port and an audio jack. That would be really cool to get in there, but it's a bit pricy and I didn't have enough spare time to invest in that.

Haven't tried it, but there should be space enough, especially if you remove the case and replace the cable with thinner leads like the enamelled wire I'm using.

People I called told me that the sound quality during calls doesn't change with the keyboard on. I still often remove the keyboard during calls, because it's a bit more comfortable.

**Reply by pfundstein** [reply]
Coule you please share us the link of the splitter device you're mentioning above? I'd like to have a look at it.

Audio jack sounds great ... Now we just need to also bring in an SD card slot and it has all the features I want haha

**Reply by Dakkaron** [reply]
Here's one with USB OTG, audio jack and charging: https://www.amazon.com/UWECAN-Headphone-Charging-Splitter-Compatible/dp/B0BKPWVMCP

This one here does USB OTG, SD, MicroSD and charging: https://www.amazon.com/Charging-JSAUX-Adapter-Compatible-MacBook/dp/B0C6PCVNNY

I don't know whether they'd actually fit, but they look as if they could, especially when removing the housing and shortening the cable.

**Reply by pfundstein** [reply]
It looks like it could be hard to remove the plastic housing, but worth a try I guess.

**Reply by Dakkaron** [reply]
Yeah, it would be better to find a naked module that only contains the chips, maybe even without connectors. But so far I haven't found anything.

And both pass-through chargin and USB audio aren't trivial and I don't have too much spare time, so I don't know whether I'll ever integrate that into the Fairberry directly.

**Reply by pfundstein** [reply]
This could also be an option to start with: https://www.amazon.de/USB-Splitter-Verteiler-Ladeanschluss-Chromecast-Gray/dp/B0CHF3RV3L/ref=sr_1_2?
